### ISSUE TITLE
bash editor blocks the build 

### ISSUE THREAD

**Issue by alef75** [de-jcup/eclipse-bash-editor#214]
This is a continuation of issue #206:
After updating to basheditor 2.4.0 the behaviour is better then 2.3.0:

ver 2.2 No Issue
ver 2.3 Eclipse freeze completely if script is opened or bash editor prefences is accessed
ver 2.4 Eclise UI work all the way, file can be edited.., but build blocks forever  and no other build can be launched

I've tested with both eclipse 2020-03 and 2020-06: the behaviour is the same.

The issue is related in some way with the build command:
as note I can tell that the UI freeze noted on ver 2.3.0 start to appear after a build:
if no build is issued  than no issue appears, after the first builds occurs, start also the issues. 

**behavior OK** 
1) No bash file opened, no access to basheditor preferences:
2) launch a build
3) build start and finish 
4) build process disappears from the eclipse progress view
4) build can be launched again

**behavior NOK ** 
1) bash file opened
2) Install bash editor model and task support appears on the eclipse progress view
3) launch a build
4) build start and reach the end but can't terminate
5) build process get stuck in the eclipse progress view
4) build can't be launched again : all the subsequent build command accumulate on the eclipse progress view forever and can't be terminated.
5) exit from eclipse
6) eclipse clode the ide and show a Progess information window "The user operation is waiting for background work to complete" and stay there forever
7) force close with killall java
8) eclipse exit.
9) reopen eclipse 
10) as the bash script file was opened on the last session , eclise open open up with the file open
11)Install bash editor model and task support appears on the eclipse progress view
12) launch a build
13) build start and reach the end but can't terminate
14) build process get stuck in the eclipse progress view
15) build can't be launched again : all the subsequent build command accumulate on the eclipse progress view forever and can't be terminated.
16) close the  bash script file
17) exit from eclipse
18) eclipse clode the ide and show a Progess information window "The user operation is waiting for background work to complete" and stay there forever
19) force close with killall java
20) eclipse exit.
21) reopen eclipse
22) no bash script file are opened.
23) from here all work right as described in **behavior OK**, but as soon I open a bash file we come back to **behavior NOK ** 

![basheditor](https://user-images.githubusercontent.com/37915977/89388452-57236580-d704-11ea-9eb8-4813cd2564b8.png)
![basheditor_exit](https://user-images.githubusercontent.com/37915977/89388478-61ddfa80-d704-11ea-86d4-c1991cbb01ea.png)

**Comment by de-jcup** [comment]
Hello @alef75 ,

pleasy try out to **disable** `Shared model build enabled` option in preferences: 

![image](https://user-images.githubusercontent.com/11644753/91107259-208e9a00-e675-11ea-8677-cb1aebc6f635.png)

Then try again - this should avoid building the shared model for bash editor. 

If this does not help, try to disable task support as well:

![image](https://user-images.githubusercontent.com/11644753/91107397-811dd700-e675-11ea-8329-dd6ebf6f3334.png)

Normally there should no longer be a build.

**Comment by de-jcup** [comment]
The steps above are only workarounds - doing them will reduce features of bash editor (but at least your build problem should not appear).

Missing features:
- Hyperlinking
- References
- TODO/FIXME tasks

**Comment by de-jcup** [comment]
It is difficult to help you, because I have not same setup/sources/workpaces as you. So please do me a favor:

- install a fresh CDT from https://www.eclipse.org/downloads/packages/ and after this install only the bash editor plugin and do not change settings 
- noe open you workspace/create a new one etc.
- open the bash files you want to edit and inspect build behaviour

If this still freezes your IDE I should be able to reproduce the situation (maybe I need a reduced example project from you which does always freeze...)

If its not freezing... Other plugins could be responsible. If have seen in your screenshots from #206 you are using some special dark theme.
- please add your additional plugins etc. step by step and test after every install if the freeze is active or not

**Comment by alef75** [comment]
Sorry for late response.
I think I will make the tests you ask me next week.
I'll let you know the results.
In the meantime are there any way to install basheditor v 2.2.3 ? in eclipse (this version work right in my enviroment)
thank you

**Comment by de-jcup** [comment]
In Mainmenu: `Window -> Install new software`

(If necessary you have to add new update site https://dl.bintray.com/de-jcup/basheditor - but should be already there)

Then disable "Show only latest...." - as marked in next picture: 
![image](https://user-images.githubusercontent.com/11644753/94110586-27aaf280-fe43-11ea-862c-5104883f3d47.png)

Now you should be able to install 2.2.3

**Comment by alef75** [comment]
Thank you for your time
I've disabled both shared modules and todo task, but, I've got the same block beaviour.
Thanks to tour suggestion I've installed sucessfully the 2.2.3 version also in eclipse 2020-09:
with 2.2.3 all works right.

For the test I made the problem is related to my project and not to workspaces in intself.
In fact if in the same workspace I close my CDT big project, and open a base CDT hello world project with some bash files, all work right.
In the next days / week I will try to isolate the problem so that I can send to you the workspace/project to help you to replicate.

For now having the V 2.2.3 installed in my new eclipse I can use it to code bash script
